### PR TITLE
fix: cannot block the current thread from within a runtime

### PR DESCRIPTION
**Pull request by SyMind**
## Require Documentation?

- [x] No
- [ ] Yes, the corresponding rspack-website PR is \_\_

### CODE REVIEW

**Review by JSerFeng — DISMISSED** [review]

File: crates/rspack_plugin_devtool/src/lib.rs [hunk 1]
```diff
@@ -564,7 +569,7 @@ impl SourceMapDevToolPlugin {
     };
 
     return match module_filename_template {
-      ModuleFilenameTemplate::Fn(f) => f(ctx),
+      ModuleFilenameTemplate::Fn(f) => f(ctx).await.expect("TODO:"),
```
Comment (h-a-n-a) [inline crates/rspack_plugin_devtool/src/lib.rs:572]: Should've return a `Result` here.

**Comment by h-a-n-a** [comment]
Have you added a test for function type value?

**Comment by SyMind** [comment]
@h-a-n-a Yes, you can see here: `packages/rspack/tests/configCases/source-map/resource-path/webpack.config.js`.

But no test case covering multiple compilers.

**Comment by h-a-n-a** [comment]
!eco-ci

**Comment by h-a-n-a** [comment]
> @h-a-n-a Yes, you can see here: `packages/rspack/tests/configCases/source-map/resource-path/webpack.config.js`.
> 
> But no test cases cover multiple compilers.

Could you please confirm if this case covers https://github.com/web-infra-dev/rspack-ecosystem-ci/actions/runs/7620597663/job/20755597690#step:8:2453

**Comment by SyMind** [comment]
@h-a-n-a I have built rspack, link it in rsbuild repository, and run rsbuild test on my computer.

**Review by h-a-n-a — APPROVED** [review]

**Comment by SyMind** [comment]
!bench

**Pull request merged**